templates/decay_amplitudes.cpp.in

```diff
@@ -0,0 +1,262 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+/**
+ * @todo implement formulas for most general forms of amplitudes
+ */
+
+#include "@ModelName@_decay_amplitudes.hpp"
+#include "wrappers.hpp"
+#include <iomanip>
+
+namespace flexiblesusy {
+
+double Decay_amplitude_SSS::square() const
```

Comment:
It appears to me that most functions in this file are generic and not model-specific. Is this correct? If so, I'd propose to move them to the `src/` directory. This will reduce overall code size, improves development/compile time and probably makes it more clear for a user/developer where to find information about these functions.

Reply:
Same applies probably to the corresponding header file, `decay_amplitudes.hpp.in`.